lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -443,13 +472,21 @@ impl SegmentEntry for ProxySegment {
             LockedSegment::Original(raw_segment) => {
                 let point_offset = raw_segment.read().get_internal_id(point_id);
                 if point_offset.is_some() {
-                    was_deleted = self.deleted_points.write().insert(point_id);
+                    was_deleted = self
+                        .deleted_points
+                        .write()
+                        .insert(point_id, op_num)
+                        .is_some();
```

Comment:
Condition is accidentally inverted.

```suggestion
                        .is_none();
```

`insert` returns `None` now if it's newly inserted. It did return `true` before if something was newly inserted. We only want this to be `true` for new insertions to only consider those as actual deletions.